File: SDE/DeleteNodeOone.py
#Definition for singly-linked list.
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

def create_LL(arr):
    head  = ListNode(arr[0])
    curr = head
    for n in arr[1:]:
        curr.next = ListNode(n)
        curr  = curr.next
    return head


class Solution:
    def deleteNode(self, node):
        node.val = node.next.val
        node.next  = node.next.next

ll = create_LL([4,5,1,9])
head  = ll 
ll = ll.next

s = Solution()
s.deleteNode(ll)

while head:
    print(head.val ,end = " ")
    head = head.next

File: SDE/DeleteNodeOone.java
public class DeleteNodeOone {
    public static void main(String[] args) {
        ListNode head = new ListNode(4);
        ListNode curr = head;
        curr.next  = new ListNode(5);
        curr = curr.next;
        curr.next = new ListNode(1);
        curr= curr.next;
        curr.next = new ListNode(9);

        curr = head.next;
        DNO dno = new DNO();
        dno.deleteNode(curr);
        
        curr = head;
        while (curr != null){
            System.out.println(curr.val);
            curr = curr.next;
        }
    }
}

class DNO {
    public void deleteNode(ListNode node) {
        ListNode next = node.next;
        ListNode curr = node;
        ListNode prv = null;
        while (next != null){
            curr.val = next.val;
            next = next.next;
            prv= curr;
            curr = curr.next;
        }
        prv.next  = null;
    }
}